### DISCUSSION TITLE
Custom broadphase in C#?

### DISCUSSION THREAD

**Discussion by lucasmontec-fortis** [amerkoleci/JoltPhysicsSharp#84] [Q&A]
Hey, so a while ago I was making a voxel experiment. I was trying to add some physics to it but I realised that it's very expensive to build meshes for the chunks. Not only that, it's also the worst way since the meshes are huge and they are picked a LOT on the broadphase. The best solution for me would be to customize the broadphase (actually both, the narrow too) and include a step to filter the voxels as AABBs directly which I can make fast and efficient.

My question is, in this wrapper do you expose any API to customize or change entirely both phases?
Is my question stupid? I mean, am I murdering performance by thinking this could be done in the managed realm?

**Comment by amerkoleci** [comment]
Hi,
Do you have any example being done with JoltPhysics in C++ so I can understand?

Thanks

**Reply by lucasmontec** [reply]
Hey! accidentally posted the question on my work account. dang.
Unfortunately, no. I'll take a look if I can make a quick example.
I'm not that familiar with Jolt, but I was writing my physics solution in C#.
What I did was use a grid (an octree or sparse tree would also work) to do the general broad phase for all of the objects, and then a custom step to deal with world collisions (since I don't need to add each voxel to the grid).  The grid phase would exclude object-to-object collisions (all of my objects were AABBs) and the world phase would exclude far voxels, considering only the local region for each object.
Adding this extra broad phase step was much faster and cheaper than trying to add extra code to solve collisions after a native physics engine was done for each frame.
